src/Views/FirmwareReleaseView.vala

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2021 elementary, Inc. (https://elementary.io)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ * Authored by: Marius Meisenzahl <mariusmeisenzahl@gmail.com>
+ */
+
+public class About.FirmwareReleaseView : Gtk.Grid {
+    private Gtk.Button update_button;
+    private Gtk.Label title_label;
+    private Gtk.Label summary_label;
+    private Gtk.Label description_label;
+    private Gtk.Label version_value_label;
+    private Gtk.Label vendor_value_label;
+    private Gtk.Label size_value_label;
+    private Gtk.Label install_duration_value_label;
+
+    public signal void back ();
+    public signal void update (Fwupd.Device device, Fwupd.Release release);
+
+    public void update_view (Fwupd.Device device, Fwupd.Release release) {
+        switch (release.flag) {
+            case Fwupd.ReleaseFlag.IS_UPGRADE:
+                if (release.version == device.version) {
+                    update_button.label = _("Up to date");
+                    update_button.sensitive = false;
+                    break;
+                }
+
+                update_button.label = _("Update");
+                update_button.sensitive = true;
+
+                update_button.clicked.connect (() => {
+                    update (device, release);
+                });
+
+                break;
+            default:
+                update_button.label = _("Up to date");
+                update_button.sensitive = false;
+                break;
+        }
+
+        title_label.label = device.name;
+        summary_label.label = release.summary;
+        description_label.label = release.description;
+        version_value_label.label = release.version;
+        vendor_value_label.label = release.vendor;
+        size_value_label.label = Formatter.bytes_to_string (release.size);
+        install_duration_value_label.label = Formatter.seconds_to_string (release.install_duration);
+
+        show_all ();
+    }
+
+    construct {
+        orientation = Gtk.Orientation.VERTICAL;
+        row_spacing = 8;
```

Comment:
We should use margins on the widgets we want to have margins instead of this spacing.

```suggestion
```